### PR TITLE
deprecate header_style

### LINKED ISSUE #4205: Deprecate `header_style`
We should drop the `header_style` option in https://github.com/sonata-project/SonataAdminBundle/blob/3.x/Resources/views/CRUD/base_list.html.twig#L62, because styling should be done with CSS, not in the html DOM.

### PR DESCRIPTION
**Pull request by osavchenko**
I am targeting this branch, because it's related to the current version and add the trigger for the deprecated option.

Closes #4205

## Changelog
```markdown
### Deprecated
- `header_style` option
```

### CODE REVIEW

File: UPGRADE-3.x.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ UPGRADE 3.x
 If you extended that controller, you should split your extended controller and
 extend the corresponding classes in `SonataAdminBundle\Action\`.
 
+## Deprecated `header_style`
```
Comment (OskarStark) [inline UPGRADE-3.x.md]:
```suggestion
## Deprecated `header_style` option
```

File: src/Resources/views/CRUD/base_list.html.twig [hunk 2]
```diff
@@ -79,6 +79,9 @@ file that was distributed with this source code.
                                             {% endif %}
 
                                             {% spaceless %}
+                                                {% if field_description.options.header_style is defined %}
+                                                    {% deprecated 'The "header_style" is deprecated, please, use "header_class" instead.' %}
```
Comment (kunicmarko20) [inline src/Resources/views/CRUD/base_list.html.twig]: Did you check if this works? Since it was merged after the last tagged release

Comment (OskarStark) [inline src/Resources/views/CRUD/base_list.html.twig]:
```suggestion
                                                    {% deprecated 'The "header_style" option is deprecated, please, use "header_class" instead.' %}
```

Comment (kunicmarko20) [reply src/Resources/views/CRUD/base_list.html.twig]: Also in the docs, it says from 2.6 and latest version is 2.5

Comment (osavchenko) [reply src/Resources/views/CRUD/base_list.html.twig]: Oh yes, you are totally right! Moved check to php code

File: UPGRADE-3.x.md [hunk 3]
```diff
@@ -6,6 +6,10 @@ UPGRADE 3.x
 If you extended that controller, you should split your extended controller and
 extend the corresponding classes in `SonataAdminBundle\Action\`.
 
+## Deprecated `header_style` option
+
+If you need to style headers prefer to use CSS classes and not in the html DOM. In this case please use `header_class` option.
```
Comment (greg0ire) [inline UPGRADE-3.x.md]: Please wrap this to 80 chars like above

**Comment by greg0ire** [comment]
Thanks @osavchenko !

**Pull request merged**